flocker/node/agents/test/test_cinder.py

```diff
@@ -61,3 +67,65 @@ def test_verify_false_ca_path(self):
             'verify_ca_path': '/a/path'
         }
         self.assertEqual(_openstack_verify_from_config(**config), False)
+
+
+class FakeCinderClient(object):
+    """"
+    Fake implementation of the cinder volume manager to use in the Cinder
+    Destroy tests.
+    Right now, we don't need a full mock of the class, we just need the
+    get not to raise any exception so the destroy volume timeout test
+    actually times out - it will be trying to get the volume in a loop
+    after deleting it, expecting a ``CinderNotFound`` exception that will
+    never happen, so we can verify that the timeout exception is rised
+    (see FLOC-1853)
+    """
+    def delete(self, volume_id):
+        """
+        A no-op delete.
+        We do not need the delete to do anything for the timeout tests.
+        """
+
+    def get(self, volume_id):
+        """
+        A no-op get.
+        Return a fake BlockDeviceVolume
+        """
+        return BlockDeviceVolume(
+            size=100, attached_to=None,
+            dataset_id=uuid4(),
+            blockdevice_id=unicode(volume_id),
+        )
+
+
+class CinderDestroyTests(SynchronousTestCase):
+    """
+    Tests for destroy_volume.
+    """
+
+    # Test added with the fix of the issue FLOC-1853
+    def test_timeout(self):
+        """
+        ``CinderBlockDeviceAPI.destroy_volume`` raises
+        ``TimeoutException`` if the volume is not removed within
+        some time (``_currentTimeout`` will define how long it will wait).
+        """
+        # Uses the fake implementation of the Cinder Volume that will
+        # be unresponsive.
+        api = CinderBlockDeviceAPI(
+            cinder_volume_manager=FakeCinderClient(),
+            nova_volume_manager=object(),
+            nova_server_manager=object(),
+            cluster_id=uuid4()
+        )
+        # Timeout = 1 because the default timeout is 300 secods, and
+        # it is a bit too long for a test that runs regularly under CI
+        returnedBlockDevice = FakeCinderClient().get(uuid4())
+        if(not isinstance(returnedBlockDevice, BlockDeviceVolume)):
```

Comment:
What's the point of checking the result of `FakeCinderClient().get`?  It's a test helper and it does return a `BlockDeviceVolume`...